packages/core/admin/admin/src/content-manager/history/components/VersionsList.tsx

```diff
@@ -1,19 +1,211 @@
+
+interface StatusData {
+  background: BoxProps['background'];
+  border: BoxProps['borderColor'];
+  text: BoxProps['color'];
+  message: MessageDescriptor;
+}
+
+interface VersionCardProps {
+  version: Contracts.HistoryVersions.GetHistoryVersions.Response['data'][number];
+  isCurrent: boolean;
+}
+
+const VersionCard = ({ version, isCurrent }: VersionCardProps) => {
+  const { formatDate, formatMessage } = useIntl();
+  const [{ query }] = useQueryParams<{ id?: string }>();
+
+  const statusData = ((): StatusData => {
+    switch (version.status) {
+      case 'draft':
+        return {
+          background: 'secondary100',
+          border: 'secondary200',
+          text: 'secondary700',
+          message: {
+            id: 'content-manager.containers.List.draft',
+            defaultMessage: 'Draft',
+          },
+        };
+      case 'modified':
+        return {
+          background: 'alternative100',
+          border: 'alternative200',
+          text: 'alternative700',
+          message: {
+            // TODO: check the translation key once D&P v5 is done
+            id: 'content-manager.containers.List.modified',
+            defaultMessage: 'Modified',
+          },
+        };
+      case 'published':
+      default:
+        return {
+          background: 'success100',
+          border: 'success200',
+          text: 'success700',
+          message: {
+            id: 'content-manager.containers.List.published',
+            defaultMessage: 'Published',
+          },
+        };
+    }
+  })();
+  const isActive = query.id === version.id.toString();
+  const author = version.createdBy && getDisplayName(version.createdBy, formatMessage);
+
+  return (
+    <Flex
+      direction="column"
+      alignItems="flex-start"
+      gap={3}
+      hasRadius
+      borderWidth="1px"
+      borderStyle="solid"
+      borderColor={isActive ? 'primary600' : 'neutral200'}
+      paddingTop={5}
+      paddingBottom={5}
+      paddingLeft={5}
+      paddingRight={5}
+      as={Link}
+      to={`?${stringify({ ...query, id: version.id })}`}
+      style={{ textDecoration: 'none' }}
+    >
+      <Flex direction="column" gap={1} alignItems="flex-start">
+        <Typography as="h3" fontWeight="semiBold">
+          {formatDate(version.createdAt, {
+            day: 'numeric',
+            month: 'numeric',
+            year: 'numeric',
+            hour: '2-digit',
+            minute: '2-digit',
+          })}
+        </Typography>
+        <Typography as="p" textColor="neutral600">
+          {formatMessage(
+            {
+              id: 'content-manager.history.sidebar.versionDescription',
+              defaultMessage:
+                '{distanceToNow}{isAnonymous, select, true {} other { by {author}}}{isCurrent, select, true { <b>(current)</b>} other {}}',
+            },
+            {
+              distanceToNow: <RelativeTime timestamp={new Date(version.createdAt)} />,
```

Comment:
Nice! Wouldn't have thought to do that 🙃

Reply:
I was surprised to see it working!